src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/QueueHealthDecider.java

```diff
@@ -0,0 +1,122 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.deciders;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.Action;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.QueueCapacity;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.AllMetrics;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.AllMetrics.ThreadPoolType;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.flow_units.ResourceFlowUnit;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries.HotClusterSummary;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries.HotNodeSummary;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries.HotResourceSummary;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.NodeKey;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.QueueRejectionClusterRca;
+
+import java.util.ArrayList;
+import java.util.List;
+
+// This is a sample decider implementation to finalize decision maker interfaces.
+// TODO: 1. Get threadpool type from HotNodeSummary or HotResourceSummary
+// TODO: 2. Read action priorities from a configurable yml
+// TODO: 3. Read current queue capacity from NodeConfigurationRca (PR #252)
+// TODO: 4. Add unit tests
+
+public class QueueHealthDecider extends Decider {
+
+    public static String NAME = "queue_health";
+
+    private QueueRejectionClusterRca queueRejectionRca;
+    List<String> actionsByUserPriority = new ArrayList<>();
+    private int counter = 0;
+
+    public QueueHealthDecider(long evalIntervalSeconds, int decisionFrequency, QueueRejectionClusterRca queueRejectionClusterRca) {
+        // TODO: Also consume NodeConfigurationRca
+        super(evalIntervalSeconds, decisionFrequency);
+        this.queueRejectionRca = queueRejectionClusterRca;
+        configureActionPriority();
+    }
+
+    @Override
+    public String name() {
```

Comment:
is there any specific reason to override this method ? The default implementation is to return getClass().getSimpleName() which will return "QueueHealthDecider" if called from this class.

Reply:
If we maintain a registry of all graph objects going forward, to allow a refactor as well as pluggable graph nodes, then the `public static NAME` field will serve as a key. For comparisons in that case, each node should return their `NAME` variable in `name()`. Hence the override